### ISSUE TITLE
Accessibility of galleries

### ISSUE THREAD

**Issue by pulsipherd** [elementor/elementor#17882]
I'm sure there's a reason that the gallery widget renders all the images as background images, but I wanted to point out the trouble this causes for users of screen readers. Standard images expose alternative text to screen readers to deliver to their users so that the blind and low-vision patrons have an equivalent experience to sighted ones. Background images, on the other hand, have no alternative text available, and the only way to give blind visitors the equivalent on a background image is to set an aria-label. It's nice that Elementor now allows for adding custom attributes, so I can add an aria-label, but only on the gallery as a whole. Trying to sum up what the total of all the images represent isn't as likely to give blind patrons the same experience as if each image could be described. Therefore, I propose three possible solutions:

1. Best would be to change the gallery from a series of div elements with background images to a series of img elements with proper exposure of the alternative text set in the WordPress back end.
2. Next best would be to give each div an aria-label attribute equal to the alternative text set in the back end, if it is set in the back end, and give it no aria-label if there's no alt text set.
3. Least convenient, though still acceptable, would be to have some way in the gallery widget to set an aria-label for each image on an individual basis.

For now I'm left with trying to describe the gallery as a whole in a unified aria-label. Not ideal.

~Captain Accessible, SeaMonster Studios

**Comment by rami-elementor** [comment]
Unfortunately, the "Basic Gallery" widget is based in WordPress `[gallery]` shortcode (https://codex.wordpress.org/Gallery_Shortcode).  Elementor can't change the shortcode, it can simple use it as is.

**Comment by Bovelett** [comment]
@rami-elementor I understand what you are saying. But I would like to know if Elementor plans to create a gallery widget that IS accessible. Elementor markets itself as a no-coding solution. That the WordPress shortcode creates inaccessible output is no reason for Elementor to not do better.

**Comment by DavidGMiles** [comment]
> Unfortunately, the "Basic Gallery" widget is based in WordPress `[gallery]` shortcode (https://codex.wordpress.org/Gallery_Shortcode). Elementor can't change the shortcode, it can simple use it as is.

Of course they can - you simply write your own function, remove the original and put your new function in it's place

remove_shortcode('gallery');
add_shortcode('gallery', 'my_gallery_shortcode');

I've done this a million times - it has the advantage that it is seamless - all old galleries move to the new presentation style

**Comment by rami-elementor** [comment]
@DavidGMiles changing the structure will change the HTML tags, which may cause backwards compatibility issues for websites using custom CSS/JS.

**Comment by rami-elementor** [comment]
Update: Both "**Basic Gallery**" and "**Gallery**" now have `role="img"` and `aria-label=""` with image ALT text.
